From 21fd39735543f6a28350a463a880bb4c3e19510f Mon Sep 17 00:00:00 2001
From: Sergio Pena <sergio.pena@cloudera.com>
Date: Tue, 12 Jul 2016 13:30:15 -0500
Subject: [PATCH 0670/1164] CDH-42201: HIVE-13704: Don't call DistCp.execute()
 instead of DistCp.run() (Sergio Pena, reviewed by
 Ashutosh Chauhan)

Change-Id: I4694e7ce370023405af0eb9ac48a793df4a2240f
---
 .../apache/hadoop/hive/shims/Hadoop23Shims.java    |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
index 6d8f285..d1c0592 100644
--- a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
+++ b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
@@ -1143,10 +1143,20 @@ public boolean runDistCp(Path src, Path dst, Configuration conf) throws IOExcept
     options.setSyncFolder(true);
     options.setSkipCRC(true);
     options.preserve(FileAttribute.BLOCKSIZE);
+
+    // Creates the command-line parameters for distcp
+    String[] params = {"-update", "-skipcrccheck", src.toString(), dst.toString()};
+
     try {
       DistCp distcp = new DistCp(conf, options);
-      distcp.execute();
-      return true;
+
+      // HIVE-13704 states that we should use run() instead of execute() due to a hadoop known issue
+      // added by HADOOP-10459
+      if (distcp.run(params) == 0) {
+        return true;
+      } else {
+        return false;
+      }
     } catch (Exception e) {
       throw new IOException("Cannot execute DistCp process: " + e, e);
     }
-- 
1.7.9.5

